include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -2686,6 +2766,181 @@ MGTwoLevelTransfer<dim, LinearAlgebra::distributed::Vector<Number>>::
+
+    for (const auto &cell : tria_fine.active_cell_iterators())
+      if (cell->is_locally_owned())
+        add_indices(cell, cell_id_translator, is_fine);
+  }
+
+
+
+  template <int dim, int spacedim>
+  LinearAlgebra::distributed::Vector<double>
+  FirstChildPolicy<dim, spacedim>::partition(
+    const Triangulation<dim, spacedim> &tria_coarse_in) const
+  {
+    const auto communicator = tria_coarse_in.get_communicator();
+
+    const internal::CellIDTranslator<dim> cell_id_translator(n_coarse_cells,
+                                                             n_global_levels);
+
+    IndexSet is_coarse(cell_id_translator.size());
+
+    for (const auto &cell : tria_coarse_in.active_cell_iterators())
+      if (cell->is_locally_owned())
+        is_coarse.add_index(cell_id_translator.translate(cell));
+
+    std::vector<unsigned int> owning_ranks_of_ghosts(is_coarse.n_elements());
+    {
+      Utilities::MPI::internal::ComputeIndexOwner::ConsensusAlgorithmsPayload
+        process(
+          is_fine, is_coarse, communicator, owning_ranks_of_ghosts, false);
+
+      Utilities::MPI::ConsensusAlgorithms::Selector<
+        std::pair<types::global_cell_index, types::global_cell_index>,
+        unsigned int>
+        consensus_algorithm(process, communicator);
+      consensus_algorithm.run();
+    }
+
+    const auto tria =
+      dynamic_cast<const parallel::TriangulationBase<dim, spacedim> *>(
+        &tria_coarse_in);
+
+    Assert(tria, ExcNotImplemented());
+
+    LinearAlgebra::distributed::Vector<double> partition(
+      tria->global_active_cell_index_partitioner().lock());
+
+    for (const auto &cell : tria_coarse_in.active_cell_iterators())
+      if (cell->is_locally_owned())
+        partition[cell->global_active_cell_index()] =
+          owning_ranks_of_ghosts[is_coarse.index_within_set(
+            cell_id_translator.translate(cell))];
+
+    partition.update_ghost_values();
+
+    return partition;
+  }
+
+
+  template <int dim, int spacedim>
+  MinimalGranularityPolicy<dim, spacedim>::MinimalGranularityPolicy(
+    const unsigned int n_min_cells)
+    : n_min_cells(n_min_cells)
+  {}
+
+
+
+  template <int dim, int spacedim>
+  LinearAlgebra::distributed::Vector<double>
+  MinimalGranularityPolicy<dim, spacedim>::partition(
+    const Triangulation<dim, spacedim> &tria_in) const
+  {
+    const auto tria =
+      dynamic_cast<const parallel::TriangulationBase<dim, spacedim> *>(
+        &tria_in);
+
+    Assert(tria, ExcNotImplemented());
+
+    LinearAlgebra::distributed::Vector<double> partition(
+      tria->global_active_cell_index_partitioner().lock());
+
+    unsigned int n_locally_owned_active_cells = 0;
+    for (const auto &cell : tria_in.active_cell_iterators())
+      if (cell->is_locally_owned())
+        ++n_locally_owned_active_cells;
+
+    if (Utilities::MPI::min(n_locally_owned_active_cells,
+                            tria_in.get_communicator()) > n_min_cells)
+      return {};
```

Comment:
Can you comment what this case means, namely that *all* MPI ranks have more than the minimum specified so we can just use the given ownership, and that can be obtained with the empty vector? I had to think for a few minutes (!) to grasp what is going on here.

Reply:
Done. I have also rewritten the code here yesterday (a bit OpenMP-like scheduling).